src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/ExternalDataBaseEntityFmsDTOValidator.java

```diff
@@ -0,0 +1,146 @@
+			externalDBEntity.setSecondaryIdentifiers(new ArrayList<>(existingSecondaryIds));
+		}
+
+		List<Long> currentXrefIds;
+		if (externalDBEntity.getCrossReferences() == null) {
+			currentXrefIds = new ArrayList<>();
+		} else {
+			currentXrefIds = externalDBEntity.getCrossReferences().stream().map(CrossReference::getId).collect(Collectors.toList());
+		}
+
+		List<Long> mergedXrefIds = null;
+		List<CrossReference> mergedCrossReferences = new ArrayList<>();
+		if(CollectionUtils.isNotEmpty(dto.getCrossReferences())) {
+			mergedCrossReferences = crossReferenceService.getMergedFmsXrefList(dto.getCrossReferences(), externalDBEntity.getCrossReferences());
+			mergedXrefIds = mergedCrossReferences.stream().map(CrossReference::getId).collect(Collectors.toList());
+			externalDBEntity.setCrossReferences(mergedCrossReferences);
+		} else {
+			mergedXrefIds = new ArrayList<>();
+			externalDBEntity.setCrossReferences(null);
+		}
+
+		Map<String, CrossReference> mergedXrefUniqueIdsMap = new HashMap<>();
+		for(CrossReference mergedXref : mergedCrossReferences) {
+			mergedXrefUniqueIdsMap.put(crossReferenceService.getCrossReferenceUniqueId(mergedXref), mergedXref);
+		}
+
+		if(externalDBEntity.getPreferredCrossReference() != null) {
+			if(dto.getPreferredCrossReference() != null) {
+				CrossReference incomingPreferredXref = createNewCrossReference(dto.getPreferredCrossReference());
+				String incomingXrefUniqueId = crossReferenceService.getCrossReferenceUniqueId(incomingPreferredXref);
+				String currentXrefUniqueId = crossReferenceService.getCrossReferenceUniqueId(externalDBEntity.getPreferredCrossReference());
+
+				if(!incomingXrefUniqueId.equals(currentXrefUniqueId)){
+					externalDBEntity.setPreferredCrossReference(null);
+					if(mergedXrefUniqueIdsMap.containsKey(incomingXrefUniqueId)) {
+						externalDBEntity.setPreferredCrossReference(mergedXrefUniqueIdsMap.get(incomingXrefUniqueId));
+					} else {
+						externalDBEntity.setPreferredCrossReference(crossReferenceDAO.persist(incomingPreferredXref));
+					}
+				}
+			} else {
+				externalDBEntity.setPreferredCrossReference(null);
+			}
+		} else {
+			if(dto.getPreferredCrossReference() != null) {
+				CrossReference incomingPreferredXref = createNewCrossReference(dto.getPreferredCrossReference());
+				String incomingXrefUniqueId = crossReferenceService.getCrossReferenceUniqueId(incomingPreferredXref);
+				if(mergedXrefUniqueIdsMap.containsKey(incomingXrefUniqueId)) {
+					externalDBEntity.setPreferredCrossReference(mergedXrefUniqueIdsMap.get(incomingXrefUniqueId));
+				} else {
+					externalDBEntity.setPreferredCrossReference(crossReferenceDAO.persist(incomingPreferredXref));
+				}
+			}
+		}
+
+	externalDataBaseEntityDAO.persist(externalDBEntity);
+
+	for (Long currentId : currentXrefIds) {
+		if (!mergedXrefIds.contains(currentId)) {
+			crossReferenceDAO.remove(currentId);
+		}
+	}
+
+	if(externalDBEntityResponse.hasErrors()) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.
<!-- __reviewdog__:ChA1YTNjMjcwYzE5MTJjMzFmEgpjaGVja3N0eWxl -->